File: Codeforces/div821/B.py

def inp():
    return list(map(int, input().split()))

for _ in range(int(input())):
    n, x, y = inp()
    if x * y != 0:
        print(-1)
        continue
    x = max(x, y)
    if x == 0 or (n - 1) % x != 0:
        print(-1)
        continue
    i = 1
    while i < n:
        for j in range(x):
            print(i + 1, end=' ')
        i += x
    print()

File: Codeforces/div821/B.cpp

#include <iostream>
using namespace::std;

int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		int n, x, y;
		cin >> n >> x >> y;
		if (x * y != 0)
		{
			cout << "-1\n";
			continue ;
		}
		x = max(x, y);
		if (x == 0 or (n - 1) % x != 0)
		{
			cout << "-1\n";
			continue ;
		}
		int i=1;
		while (i < n)
		{
			for (int j=0; j < x; ++j)
				cout << i + 1 << ' ';
			i += x;
		}
		cout << endl;
	}
}

